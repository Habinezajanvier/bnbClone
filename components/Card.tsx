import Image from "next/image";
import { AiFillStar } from "react-icons/ai";
import { BsDot } from "react-icons/bs";
import hotelImage from "../assets/hotel_picture.jpeg";
import styles from "../styles/Home.module.css";
import AliceCarousel, { DotsItem } from "react-alice-carousel";

export default function Card() {
  const handleDragStart = (e) => e.preventDefault();

  const items = [
    <Image
      src={hotelImage}
      onDragStart={handleDragStart}
      role="presentation"
    />,
    <Image
      src={hotelImage}
      onDragStart={handleDragStart}
      role="presentation"
    />,
    <Image
      src={hotelImage}
      onDragStart={handleDragStart}
      role="presentation"
    />,
  ];

  return (
    <div className={styles.card}>
      <div className={styles.imageHolder}>
        {/* <Image src={hotelImage} alt="hotel_picture" /> */}
        <AliceCarousel
          mouseTracking
          items={items}
          controlsStrategy="alternate"
          // renderDotsItem={(e: DotsItem) => (
          //   <div className={styles.carouselDot}>
          //     <BsDot />
          //   </div>
          // )}
        />
      </div>
      <div className={styles.descriptions}>
        <div>
          <p className={styles.name}>Cape town, South Africa</p>
          <p className={styles.descriptionText}>Hosted by Louise</p>
          <p className={styles.descriptionText}>Jan 10-15</p>
          <p className={styles.price}>
            <span>$49</span> night
          </p>
        </div>
        <div className={styles.rating}>
          <AiFillStar size="1rem" />
          <p>5.0</p>
        </div>
      </div>
    </div>
  );
}
