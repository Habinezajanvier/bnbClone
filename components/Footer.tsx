import styles from "../styles/Home.module.css";

export default function Footer() {
  return (
    <div className={styles.footer}>
      <p>copyright @hotel & co {new Date().getFullYear()}</p>
    </div>
  );
}
