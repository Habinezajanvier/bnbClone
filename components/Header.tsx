import React, { useState } from "react";
import { SiAcer } from "react-icons/si";
import { SlCup } from "react-icons/sl";
import styles from "../styles/Home.module.css";

export default function Header() {
  return (
    <div className={styles.headerComponent}>
      <SlCup />
      <p>Bed & Breakfast</p>
    </div>
  );
}
