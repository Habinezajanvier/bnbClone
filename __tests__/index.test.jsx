import { render, screen } from "@testing-library/react";
import Home from "../pages/index";
import "@testing-library/jest-dom";
import Card from "../components/Card";
import Header from "../components/Header";
import Footer from "../components/Footer";

describe("Home", () => {
  it("render single hotel card", () => {
    render(<Card />);

    const element = screen.getByRole("img", { name: "hotel_picture" });
    expect(element).toBeInTheDocument();
  });

  it("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
