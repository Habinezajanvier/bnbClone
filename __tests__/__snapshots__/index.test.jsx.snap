// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <main
    class="main"
  >
    <h1
      class="text-3xl font-bold underline"
    >
      Hello world!
    </h1>
  </main>
</div>
`;
